pattern/constraint/thing/IIDConstraint.java

```diff
@@ -87,4 +88,9 @@ public String toString() {
         syntax.append(Bytes.bytesToHexString(iid));
         return syntax.toString();
     }
+
+    @Override
+    protected IIDConstraint clone(ConstraintCloner cloner) {
+        return cloner.cloneVariable(owner).iid(iid);
+    }
```

Comment:
This may not be favourable in terms of naming/architecture, as we already have the style below for using VariableCloner. I didn't spot that we had this for a while though as it's not called `clone`. Feels like `clone` is more to the point to me. Can't decide if a static method is better. Thoughts?
```
static HasConstraint of(ThingVariable owner, HasConstraint clone, VariableCloner cloner) {
        return new HasConstraint(owner, cloner.clone(clone.attribute()));
    }
```

Reply:
It's an interesting deliniation - I think I prefer having a `clone()` method, over an overloaded `of`. However, it now needs to be made super clear this isn't a simple 'cut/copy' operation that is disconnected from any conjunction or graph. As such, I'd probably end up renaming the `ConstraintCloner` to `ConjunctionCloner`, and the the `clone` method is then really `cloneInto`?

To follow that line of reasoning, then the `ConjunctionCloner` is actually a `ConjunctionCloneBuilder`! Further, it seems to imply that `Cloner` should be an inner class of `Conjunction`? 

The API is then:
`Conjunction.Cloner.cloneExactly(Set<Constraint> subgraph)` - which aligns really well with the terminology you have in the `Retrievable` constructor. The equivalent of the `.build()` method that would normally exist on a Builder, here would return a new `conjunction` -- eg. exactly as you use in `Retrievable`!

The names and stuff aren't super there yet in the above paragraphs, but I think you get the gist of what I'm trying to convey